backend/src/main/java/com/jobwait/security/UUIDJumbler.java

```diff
@@ -0,0 +1,33 @@
+package com.jobwait.security;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class UUIDJumbler {
+    private static String bytesToHex(byte[] hash) {
+        StringBuilder hexString = new StringBuilder(2 * hash.length);
+        for (int i = 0; i < hash.length; i++) {
+            String hex = Integer.toHexString(0xff & hash[i]);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
+
+    // NoSuchAlgorithmException should never throw because the MessageDigest
+    // algorithm exists...
+    public static String jumbleUUID(String someID) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA3-256");
+            byte[] hashbytes = digest.digest(
+                    someID.getBytes(StandardCharsets.UTF_8));
+            final String sha3Hex = bytesToHex(hashbytes);
+            return sha3Hex;
+        } catch (NoSuchAlgorithmException exception) {
+            throw SecurityFaults.GenericAuthenticationFault();
+        }
+    }
```

Comment:
On that note, why do we need a jumbler? Is it because someID returned from linkedin isn't a UUID so we have to convert it to one? If that's the case, why are we using UUID in the database, why not just a string or something?

Reply:
This is part of the discussion we had so long ago where I got confused about the exact same thing.

Let me try to clear up some stuff. The jumbler is exactly because of what you say, the someID we get from LinkedIn is roughly a string of length 12 (you can probably find the example I pasted into our chats a couple of times, its very short). Here we simply hash it into SHA3-256. So yes, the ID is not a UUID. 

We use UUID in the db because its auto-generated and we want to identify the user. The authHash that is generated by this jumbler is solely for LinkedIn purposes. That is at least what I rationalized from our discussions and memory. Let me know if I missed something.

Reply:
my confusion is: if this is returning a SHA3hex, and the input parameter is not a UUID, why is UUID in the function name?